src/VirtualClient/VirtualClient.Monitors/AmdSmiMonitor.cs

```diff
@@ -22,27 +25,59 @@ namespace VirtualClient.Monitors
     /// </summary>
     public class AmdSmiMonitor : VirtualClientIntervalBasedMonitor
     {
+        /// <summary>
+        /// Name of Metric subsystem.
+        /// </summary>
+        protected const string Metric = "metric";
+
+        /// <summary>
+        /// Name of XGMI subsystem.
+        /// </summary>
+        protected const string XGMI = "xgmi";
+
+        private ISystemManagement systemManagement;
+        private IFileSystem fileSystem;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="AmdSmiMonitor"/> class.
         /// </summary>
         public AmdSmiMonitor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
             : base(dependencies, parameters)
         {
+            this.systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            this.fileSystem = this.systemManagement.FileSystem;
+        }
+
+        /// <summary>
+        /// AMDSMI Subsystem Name.
+        /// </summary>
+        public string Subsystem
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(AmdSmiMonitor.Subsystem), out IConvertible subsystem);
+                return subsystem?.ToString();
+            }
         }
 
         /// <inheritdoc/>
         protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
         {
-            switch (this.Platform)
+            if (this.Platform == PlatformID.Win32NT || this.Platform == PlatformID.Unix)
             {
-                case PlatformID.Win32NT:
-                    await this.QueryGpuAsync(telemetryContext, cancellationToken)
-                        .ConfigureAwait(false);
-                    break;
+                var tasks = new List<Task>();
 
-                case PlatformID.Unix:
-                    // not supported at the moment
-                    break;
+                if (this.Subsystem == AmdSmiMonitor.Metric)
+                {
+                    tasks.Add(this.QueryGpuMetricAsync(telemetryContext, cancellationToken));

```

Comment:
you don't have to use the tasks and Task.WhenAll if all you do is picking which one of the two methods to use

Reply:
I have done this in order for them to run parallelly if both enabled.